### PR TITLE
Fix venv instructions

### PR DESCRIPTION
**Pull request by nathan-chan-brt**
## Description

Fixes issue identified in https://discuss.scientific-python.org/t/encountering-errors-following-start-of-development-process-in-scikit-image-contributing-rst/632/2. Updates the instructions for using venv+pip to ensure virtualenv created and run outside the source tree.

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -51,11 +51,12 @@ Here's the long and short of it:
 
        ::
 
-         # Create a virtualenv named ``skimage-dev`` that lives in the directory of
-         # the same name
-         python -m venv skimage-dev
+         # Create a virtualenv named ``skimage-dev`` that lives in a dedicated directory outside
+         # the source tree. One suggestion below is to create an ~/envs folder:
```
Comment (mkcor) [inline CONTRIBUTING.rst:55]:
```suggestion
         # Create a virtualenv named ``skimage-dev`` that lives outside of the repository.
         # One convention is to use an ``envs`` directory under the home directory:
```

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -51,11 +51,12 @@ Here's the long and short of it:
 
        ::
 
-         # Create a virtualenv named ``skimage-dev`` that lives in the directory of
-         # the same name
-         python -m venv skimage-dev
+         # Create a virtualenv named ``skimage-dev`` that lives outside of the repository.
+         # One convention is to use an ``envs`` directory under the home directory:
+         mkdir -p ~/envs
```
Comment (stefanv) [inline CONTRIBUTING.rst:56]:
```suggestion
         mkdir ~/envs
```

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -51,11 +51,12 @@ Here's the long and short of it:
 
        ::
 
-         # Create a virtualenv named ``skimage-dev`` that lives in the directory of
-         # the same name
-         python -m venv skimage-dev
+         # Create a virtualenv named ``skimage-dev`` that lives outside of the repository.
+         # One convention is to use an ``envs`` directory under the home directory:
```
Comment (stefanv) [inline CONTRIBUTING.rst:55]:
```suggestion
         # One common convention is to place it inside an ``envs`` directory under your home directory:
```

**Review by mkcor — APPROVED** [review]

**Comment by stefanv** [comment]
Thanks very much @nathan-chan-brt!

**Pull request merged**